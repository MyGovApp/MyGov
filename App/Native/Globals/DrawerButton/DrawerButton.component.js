import React, { Component, PropTypes } from 'react'
import { Text, TouchableOpacity, Image, View } from 'react-native'
import styles from './DrawerButton.styles'

class DrawerButton extends Component {
  render () {
    const indexContainer = this.props.isIndex ? styles.indexContainer : {}
    const indexText = this.props.isIndex ? styles.indexText : {}
    const indexBorder = this.props.isIndex ? styles.indexBorder : {}

    return (
      <View style={[ styles.borderContainer, indexBorder ]}>
        <TouchableOpacity
          onPress={this.props.onPress}
          style={[ styles.container, indexContainer ]}
        >
          {this.props.icon &&
            (<Image source={this.props.icon} style={styles.icon} />)
          }
          <Text style={[ styles.text, indexText ]}>{this.props.text}</Text>
        </TouchableOpacity>
      </View>
    )
  }
}

DrawerButton.propTypes = {
  onPress: PropTypes.func.isRequired,
  text: PropTypes.string.isRequired,
  icon: PropTypes.number,
  isIndex: PropTypes.bool
}

export default DrawerButton
