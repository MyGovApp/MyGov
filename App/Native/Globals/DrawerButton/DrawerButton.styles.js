import { metrics, colors, fonts } from '../../Themes'

export default {
  borderContainer: {
    borderBottomColor: colors.frost,
    borderBottomWidth: 1,
    borderLeftColor: colors.frost,
    borderLeftWidth: 1
  },
  indexBorder: {
    borderBottomColor: colors.secondaryBlue,
    borderLeftColor: colors.secondaryBlue
  },
  container: {
    alignItems: 'center',
    backgroundColor: 'white',
    flexDirection: 'row',
    paddingLeft: 5,
    paddingRight: 5,
    justifyContent: 'space-between'
  },
  leftIconContainer: {
    justifyContent: 'flex-start'
  },
  indexContainer: {
    backgroundColor: colors.secondaryBlue
  },
  text: {
    ...fonts.style.h5,
    color: colors.black,
    marginVertical: metrics.baseMargin
  },
  indexText: {
    color: 'white'
  },
  icon: {
    height: 30,
    marginBottom: 10,
    marginLeft: 10,
    marginRight: 10,
    marginTop: 10,
    width: 30
  }
}
