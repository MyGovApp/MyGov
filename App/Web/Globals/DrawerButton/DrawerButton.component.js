import React, { Component, PropTypes } from 'react'
import s from './DrawerButton.syles.scss'

class DrawerButton extends Component {
  setup () {
    const { leftIcon, rightIcon } = this.props

    return {
      indexContainer: this.props.isIndex ? s.indexContainer : '',
      indexText: this.props.isIndex ? s.indexText : '',
      leftIcon,
      rightIcon
    }
  }

  render () {
    const { indexContainer, indexText, leftIcon, rightIcon } = this.setup()

    return (
      <div
        className={`${s.container} ${indexContainer}`}
        onClick={this.props.onPress}>

        <div className={s.leftGroup}>
          {leftIcon && (<img src={leftIcon} className={s.iconLeft} />)}
          <h3 className={`${s.text} ${indexText}`}>{this.props.text}</h3>
        </div>
        {rightIcon && (<img src={rightIcon} className={s.iconRight} />)}
      </div>
    )
  }
}

DrawerButton.propTypes = {
  text: PropTypes.string.isRequired,
  onPress: PropTypes.func,
  leftIcon: PropTypes.any,
  rightIcon: PropTypes.any,
  isIndex: PropTypes.bool
}

export default DrawerButton
