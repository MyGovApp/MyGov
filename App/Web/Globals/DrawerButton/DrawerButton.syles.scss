@import "../../themes/base";

.container {
  background-color: $backgroundWhite;
  padding: 8px;
  border-bottom: solid 1px $frost;
  border-left: solid 1px $frost;
  display: flex;
  justify-content: space-between;
}

.indexContainer {
  background-color: $secondaryBlue;
  border-left: solid 1px $secondaryBlue !important;
}

.text {
  font-size: 20px;
}

.indexText {
  color: white;
}

.iconLeft {
  height: 28px;
  padding-right: 10px;
}

.iconRight {
  height: 22px;
  padding-right: 10px;
}

.leftGroup {
  display: flex;
  align-items: center;
  flex-direction: row;
}
