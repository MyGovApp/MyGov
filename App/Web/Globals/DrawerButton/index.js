import DrawerButton from './DrawerButton.component.js'

export default DrawerButton
