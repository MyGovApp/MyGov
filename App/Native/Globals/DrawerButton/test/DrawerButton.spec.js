import 'react-native'
import React from 'react' // eslint-disable-line
import renderer from 'react-test-renderer'
import { ComponentWrapper } from './DrawerButton.setup'

describe('Native DrawerButton', () => {
  it('should render the DrawerButton component correctly', () => {
    const tree = renderer.create(ComponentWrapper()).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
