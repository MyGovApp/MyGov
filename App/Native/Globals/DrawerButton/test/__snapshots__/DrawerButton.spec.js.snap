// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Native DrawerButton should render the DrawerButton component correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomColor": "#D8D8D8",
        "borderBottomWidth": 1,
        "borderLeftColor": "#D8D8D8",
        "borderLeftWidth": 1,
      },
      Object {
        "borderBottomColor": "#466891",
        "borderLeftColor": "#466891",
      },
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#466891",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "opacity": 1,
        "paddingLeft": 5,
        "paddingRight": 5,
      }
    }
    testID={undefined}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 30,
            "marginBottom": 10,
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 10,
            "width": 30,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontFamily": "HelveticaNeue",
              "fontSize": 20,
              "marginVertical": 10,
            },
            Object {
              "color": "white",
            },
          ]
        }
      >
        Test
      </Text>
    </View>
    <Image
      source={1}
      style={
        Object {
          "height": 20,
          "marginBottom": 15,
          "marginLeft": 15,
          "marginRight": 15,
          "marginTop": 15,
          "width": 20,
        }
      }
    />
  </View>
</View>
`;
